app/validators/status_pin_validator.rb

```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
 
 class StatusPinValidator < ActiveModel::Validator
+  MAX_PINNED = (ENV['MAX_PINNED_TOOTS'].to_i - 1 || 4)
```

Comment:
Why the `- 1`?
Also, `nil.to_i` returns `0`, so `MAX_PINNED` is `-1` whenever `MAX_PINNED_TOOTS` isn't set.
This line should probably be `MAX_PINNED = (ENV['MAX_PINNED_TOOTS'] || 4).to_i` instead.

Reply:
The `- 1` is there because the default logic checks if the amount of pinned statuses is greater than the max. If it wasn't there then the maximum pinned statuses would be one greater than what was set in the config. This is also why I had changed it from `4` to `5` in the first commit, to preserve the default behavior.

Reply:
Oh. Right.
So the line should probably be `MAX_PINNED = (ENV['MAX_PINNED_TOOTS'] || 5).to_i - 1` as opposed to what I told earlier. Or the comparison changed to `>=`.

Reply:
i would argue to make it less confusing and just change the comparison